cmake/CMakeLists.txt

```diff
@@ -411,9 +415,21 @@ if (NOT (UNIX AND onnxruntime_ENABLE_PYTHON AND onnxruntime_ENABLE_TRAINING AND
   set(onnxruntime_ENABLE_TRAINING_TORCH_INTEROP OFF)
 endif()
 
+if (NOT (UNIX AND onnxruntime_USE_CUDA AND onnxruntime_ENABLE_PYTHON AND onnxruntime_ENABLE_TRAINING AND (NOT onnxruntime_BUILD_SHARED_LIB)))
+  if (onnxruntime_ENABLE_TRITON)
+    message(WARNING "onnxruntime_ENABLE_TRITON is turned OFF due to incompatible build combinations.")
```

Comment:
Which kind of incompatibilites? Any kind we cannot solve in the future?

Reply:
Same issue as onnxruntime_ENABLE_TRAINING_TORCH_INTEROP above, ORT's Win and Linux build has different logic of lib build and dependency logic, it requires more complex change to make Win build work, but normally there is no people training on Win. And this feature is for CUDA and with Python enabled only.

Reply:
Maybe that's worth adding more information in the error message to give users more insights.